### DISCUSSION TITLE
Cell 3, Dip-1, Suddenly not available

### DISCUSSION THREAD

**Discussion by x230-thinkpad** [txubelaxu/esphome-jk-bms#32] [General]
![image](https://github.com/user-attachments/assets/938ef13b-8caa-43e0-97cb-b7b498deeb6e)

there is something wrong with the components ?
only cell number 3 and DIP 1 has not available, I try used dip 2 until 8 all is okey

**Comment by marcellinusdeman** [comment]
I have the same issue. The logfile shows the voltage but it is not sent.

[15:32:36][D][jk_rs485_bms:690]: [ADDRESS: 01]  00 --> V: 3.422000V
[15:32:36][D][sensor:094]: 'rs485-bms1 cell voltage 01': Sending state 3.42200 V with 3 decimals of accuracy
[15:32:36][D][jk_rs485_bms:695]:                   --> R: 0.047000ohm
[15:32:36][D][jk_rs485_bms:690]: [ADDRESS: 01]  01 --> V: 3.422000V
[15:32:36][D][sensor:094]: 'rs485-bms1 cell voltage 02': Sending state 3.42200 V with 3 decimals of accuracy
[15:32:36][D][jk_rs485_bms:695]:                   --> R: 0.045000ohm
[15:32:36][D][jk_rs485_bms:690]: [ADDRESS: 01]  02 --> V: 3.421000V
[15:32:36][D][jk_rs485_bms:695]:                   --> R: 0.049000ohm
[15:32:36][D][jk_rs485_bms:690]: [ADDRESS: 01]  03 --> V: 3.422000V
[15:32:36][D][sensor:094]: 'rs485-bms1 cell voltage 04': Sending state 3.42200 V with 3 decimals of accuracy

**Reply by txubelaxu** [reply]
Sorry!! I know that it sounds very crazy, but I had to deactivate that cell info to get stability and avoid some reboots.

I have got know working without revoots and with rhat cell info recovered in the last version on my github.

**Reply by x230-thinkpad** [reply]
cell number 3 dip1 already normal, but have another problem

![image](https://github.com/user-attachments/assets/e5f360f6-3e85-444a-bc76-e7f8457f905a)

all values not update, 14 minutes still not update.

**Comment by txubelaxu** [comment]
Sorry!! I know that it sounds very crazy, but I had to deactivate that cell info to get stability and avoid some reboots.

I have got know working without revoots and with rhat cell info recovered in the last version on my github.

**Reply by x230-thinkpad** [reply]
so this version will no longer available ?

**Reply by txubelaxu** [reply]
It is the same version updated. 
I have not published a "zip" version. Only mods to github.

**Reply by x230-thinkpad** [reply]
but after your last update, the esp not event changes only read 1 time then not update the value again, I have try waiting until 8 hours the value stuck, then I re-install ESP still can't update

![image](https://github.com/user-attachments/assets/555b8083-40c9-4c27-8acb-f1e8ca672d50)

**Reply by txubelaxu** [reply]
I have readings in my system:

![image](https://github.com/user-attachments/assets/3ad7d48e-cffb-4bd6-91ef-51f09b7ea8a8)

I changed some time intervals as well. Maybe is about these changes.

I have to change some code to inprove the token assign method.

Thank you for your info.

**Reply by x230-thinkpad** [reply]
> t update, the esp not event changes only read 1 t

I'm so sorry friend, it still not working for me, did I do something wrong at my code after re-update again, all value appears only for a minutes then become unavailable again, did I do something wrong at my code?

here I attach my yaml file
[esp-rs485-gw.zip](https://github.com/user-attachments/files/17698856/esp-rs485-gw.zip)

![image](https://github.com/user-attachments/assets/ae328197-f054-4ce1-94ca-f4f9985481e5)

**Reply by txubelaxu** [reply]
My system has 2 BMSs only. So, I think that when >2BMSs, there should be adjust some "time" paramethers inside jk_rs485_sniffer.cpp before to compilation. Try to change these values:

![image](https://github.com/user-attachments/assets/27cdcebd-aecd-4161-84fe-de5bc847a8ea)

**Reply by marcellinusdeman** [reply]
It is working for me cell 03 is back and the system is updating. I have 3 BMS's, 1 Master and 2 Slaves. Type  JK_PB1A16S15P.

Thanks
Marcel

**Comment by marcellinusdeman** [comment]
Celebrated to early, it stopped updating after a while, so I had to roll back to the previous version.

Marcel

**Reply by x230-thinkpad** [reply]
> Celebrated to early, it stopped updating after a while, so I had to roll back to the previous version.
> 
> Marcel

Hi Marcel,
please tell me how to roll back to the previous version, im to newbie for this 😄

**Reply by marcellinusdeman** [reply]
Before I downloaded the updated software version, I made a copy of the esphome-jk-bms directory.

Marcel

**Reply by x230-thinkpad** [reply]
> Before I downloaded the updated software version, I made a copy of the esphome-jk-bms directory.
> 
> Marcel

I already longtime used this method it is deeply regretted still not working for me, it's wonderful project, but never mind I try find other way to monitor my bms 

cheers

**Reply by txubelaxu** [reply]
I changed sniffer.cpp some time parameters.

I will recover the previous ones. Meanwhile, could you try with old parameter values?

**Reply by x230-thinkpad** [reply]
> I changed sniffer.cpp some time parameters.
> 
> I will recover the previous ones. Meanwhile, could you try with old parameter values?

I already fallback all parameter still can't update, here I attach the logs and yaml

[esp-rs485-gw.zip](https://github.com/user-attachments/files/17706053/esp-rs485-gw.zip)
[logs_esp-rs485-gw_logs.zip](https://github.com/user-attachments/files/17706039/logs_esp-rs485-gw_logs.zip)

before everything is normal and update regularly every minutes for 8 bms use 1 ESP

I also already try replace the ESP32-S3 and Max485  module, the value only appears for reading 1st time after that no update, and now after a few minutes become unavailable 

![image](https://github.com/user-attachments/assets/cd21e90b-384f-485c-967c-7cd650f11539)

**Reply by txubelaxu** [reply]
umm...
If I remember OK, I changed a "buffer" size from 2048 to 1024 or 512.

Sorry, I cannot review now...

I would like to change the method I developped to pooling the bmss. It could be improve very much to avoid this (I think). Some work pending...

**Reply by x230-thinkpad** [reply]
> umm...
> If I remember OK, I changed a "buffer" size from 2048 to 1024 or 512.
> 
> Sorry, I cannot review now...
> 
> I would like to change the method I developped to pooling the bmss. It could be improve very much to avoid this (I think). Some work pending...

is okay mate,
i have temporary solution. all I need is only for monitor the bms to home assistant, the bms not comunicate with mppt/inverter devices.

so I try to used Rs485 converter directly to home assistant configuration.yaml, so far is good enough.

Cheers,

**Reply by x230-thinkpad** [reply]
> umm... If I remember OK, I changed a "buffer" size from 2048 to 1024 or 512.
> 
> Sorry, I cannot review now...
> 
> I would like to change the method I developped to pooling the bmss. It could be improve very much to avoid this (I think). Some work pending...

hi mate @txubelaxu ,

after your fallback sniffer time all value updates normally, I have monitor round 1 hour and keeps update and stable
![image](https://github.com/user-attachments/assets/14b13ead-40f5-402a-be7c-6eccc8a8e59d)

i appreciate, thank you very much
